apps/site/lib/site_web/views/trip_compare_view.ex

```diff
@@ -0,0 +1,686 @@
+  end
+
+  def format_plan_type_for_title(%{time: {:depart_at, dt}}) do
+    ["Depart at ", Timex.format!(dt, "{h12}:{m} {AM}, {M}/{D}/{YY}")]
+  end
+
+  def format_plan_type_for_title(%{time: {:error, _}}) do
+    format_plan_type_for_title(nil)
+  end
+
+  def format_plan_type_for_title(nil) do
+    time = Site.TripPlan.DateTime.round_minute(Util.now())
+    ["Depart at ", Timex.format!(time, "{h12}:{m} {AM}, {M}/{D}/{YY}")]
+  end
+
+  @spec format_minutes_duration(integer) :: String.t()
+  def format_minutes_duration(duration) do
+    case duration do
+      duration when duration >= 60 ->
+        "#{div(duration, 60)} hr #{rem(duration, 60)} min"
+
+      _ ->
+        "#{duration} min"
+    end
+  end
+
+  @spec transfer_route_name(Route.t()) :: String.t()
+  def transfer_route_name(%Route{type: type} = route) when type in [0, 1] do
+    route
+    |> Route.to_naive()
+    |> Map.get(:name)
+  end
+
+  def transfer_route_name(%Route{type: type}) do
+    SiteWeb.ViewHelpers.mode_name(type)
+  end
+
+  @doc "Add a transfer note to the trip plan view when the itinerary might have valid transit transfers (determined by Transfer.is_maybe_transfer?) that are not already accounted for in the fare calculation. Right now the only transfer accounted for in the calculated fare is subway-subway (via Transfer.is_subway_transfer?)"
+  @spec transfer_note(Itinerary.t()) :: String.t() | nil
+  def transfer_note(itinerary) do
+    itinerary.legs
+    |> Stream.filter(&Leg.transit?/1)
+    |> Stream.chunk_every(2, 1)
+    |> Enum.reject(&Transfer.is_subway_transfer?/1)
+    |> Enum.find(&Transfer.is_maybe_transfer?/1)
+    |> transfer_note_text
+  end
+
+  defp transfer_note_text(nil), do: nil
+
+  defp transfer_note_text(_) do
+    HTML.Tag.content_tag(
+      :span,
+      [
+        "Total may be less with ",
+        HTML.Tag.content_tag(:a, "transfers", href: "https://www.mbta.com/fares/transfers")
+      ]
+    )
+  end
+
+  def render_to_string(template, data) do
+    template |> render(data) |> HTML.safe_to_string()
+  end
+
+  def itinerary_map(map_data) do
+    map_data
+    |> Map.put(:tile_server_url, Application.fetch_env!(:site, :tile_server_url))
+  end
+
+  @spec itinerary_html(any, %{conn: atom | %{assigns: atom | map}, expanded: any}) :: [any]
+  def itinerary_html(itineraries, %{conn: conn, expanded: expanded}) do
+    old_count = Enum.count(itineraries, &(&1.source == "CURRENT"))
+
+    for {i, routes, map, links, itinerary_row_list, index} <-
+          Enum.zip([
+            itineraries,
+            conn.assigns.routes,
+            conn.assigns.itinerary_maps,
+            conn.assigns.related_links,
+            conn.assigns.itinerary_row_lists,
+            Stream.iterate(1, &(&1 + 1))
+          ]) do
+      tab_html =
+        "_itinerary_tab.html"
+        |> render_to_string(
+          itinerary: i,
+          index: if(index > old_count, do: index - old_count, else: index),
```

Comment:
why does subtracting this value work?

Reply:
this value goes to the itinerary header ("itinerary 1", "itinerary 2", etc) - we just want the numbering to restart for the new itineraries